test/FranchiserFactory.invariants.t.sol

```diff
@@ -44,14 +44,20 @@ contract FranchiseFactoryInvariantTest is Test {
         );
```

Comment:
this comment relates to `invariant_Franchisers_and_recalled_balance_sum_matches_total_supply`

it seems we're testing an invariant about our invariant suite rather than an invariant about franchiser. for example, we should expect a token's total supply to change via movements external to the system, but that would break this invariant. we should also expect delegators' balances to change outside of franchiser, but this articulation doesn't allow that either.

I think the invariant should be "sum of franchiser balances should equal amount funded minus amount recalled." we should keep two ghost vars, `ghost_funded` and `ghost_recalled`. we should increment `ghost_funded` by the amount franchisers are funded on `fund` or `fundMany` calls. we should increment `ghost_recalled` by the amount recalled when `recall` or `recallMany` is called (but only when recalled to an address external to the system, aka not the franchiser itself which happens in the case to unsubdelegate). now, we can reduce the franchisers, summing their balances, and assert that it's equal to `ghost_funded - ghost_recalled`.